Source/NetworkClients/WeatherClient.swift

```diff
@@ -0,0 +1,45 @@
+//  Copyright © 2016 Gavan Chan. All rights reserved.
+
+import Argo
+import Foundation
+import RxCocoa
+import RxSwift
+
+typealias WeatherProvidingClient = ClientProtocol & WeatherProviding
+
+final class WeatherClient: WeatherProvidingClient {
+
+  enum Endpoint: String {
+    case weather = "weather"
+  }
+
+  let session: URLSession = .shared
+  var base: URL { return URL(string: baseURL)! }
+
+  private let baseURL: String = "http://api.openweathermap.org/data/2.5"
```

Comment:
In the case there will be more clients in the future, it is worth noting that this `baseURL` may be extracted into some form of a superclass for `WeatherClient`, introducing some state-sharing inheritance hierarchy--unless we decide to not introduce this as a base URL string in the types themselves and prefer some form of "Constants" file.